Tests/UnitTests/TestHelpers/AsyncTestHelpers.swift

```diff
@@ -58,30 +58,58 @@ func waitUntilValue<Value>(
     return result.value
 }
 
+private struct ConditionFailedError: Error {}
+
+@available(iOS 13.0, macOS 10.15, tvOS 13.0, watchOS 6.2, *)
+func asyncWait(
+    description: String? = nil,
+    timeout: DispatchTimeInterval = defaultTimeout,
+    pollInterval: DispatchTimeInterval = defaultPollInterval,
+    file: FileString = #fileID,
+    line: UInt = #line,
+    until condition: @Sendable () async -> Bool
+) async throws {
+    try await asyncWait(
+        description: { _ in description },
+        timeout: timeout,
+        pollInterval: pollInterval,
+        file: file,
+        line: line,
+        until: { () },
+        condition: { _ in await condition() }
+    )
+}
+
 /// Verifies that the given `async` condition becomes true after `timeout`,
 /// checking every `pollInterval`.
 // Note: a better approach would be using `XCTestExpectation` and `self.wait(for:timeout:)`
 // but it doesn't seem to play well with async-await.
 // Also `toEventually` (Quick nor Nimble) don't support `async`.
 // Fix-me: remove once we can use Quick v6.x:
 @available(iOS 13.0, macOS 10.15, tvOS 13.0, watchOS 6.2, *)
-func asyncWait(
-    description: String? = nil,
+func asyncWait<T>(
```

Comment:
I've added this new overload that allows printing the last value in the description.
This allows us to see exactly what the failure was, instead of just:
```
failed - Expected 2 unfinished transactions
expected to be true, got <false>
```

"Expected true, got false" wasn't very helpful.